### PR TITLE
Fixed bug where Totem would deselect when picking a troll.

### PR DESCRIPTION
**Pull request by totony-misc**
Totem would deselect sometimes when u picked a troll, was just kinda annoying. This should fix it

### CODE REVIEW

**Comment by totony-misc** [comment]
btw, Asud isn't in Stdlib so I just hard-coded it

File: wurst/objects/units/ShopObjects.wurst [hunk 1]
```diff
@@ -34,3 +40,26 @@ public function HeroDefinition.setTotemShopInterface()
     ..setStockReplenishInterval(0)
     ..setIconGameInterface(Icons.bTNCancel)
     ..setUnitSoundSet("")
+
+@compiletime function createTotem()
+    let ABILITY_SELECT_UNITS_TOTEM = compiletime(UNIT_ID_GEN.next())
+
+    new AbilityDefinitionAlliedBuilding(ABILITY_SELECT_UNITS_TOTEM)
+    ..setActivationRadius(1, 1) // Small activation radius so it doesn't try selecting trolls
```
Comment (jonathanmclaus) [inline wurst/objects/units/ShopObjects.wurst:48]: Always indent `..` blocks.

Comment (totony-misc) [reply wurst/objects/units/ShopObjects.wurst:48]: I based it off units.wurst which doesn't indent .. blocks but I can indent them

**Review by theQuazz — COMMENTED** [review]
👏 hated that I had to have those stupid `UNIT_HERO_PICKER`s for each player

File: wurst/systems/core/ClassSelector.wurst [hunk 2]
```diff
@@ -134,21 +132,11 @@ public class ClassSelector
             ..add(UNIT_SCOUT)
             ..add(UNIT_GATHERER)
 
-    private function initializePickers()
-        let pos = spawn.getCenter()
-        for member in tribe.getMembers()
-            pickers.push(createUnit(member, UNIT_HERO_PICKER, pos.polarOffset(angle(PI), 1200), angle(0)))
-            member.panCamToTimed(totem, 0)
-            doAfter(0.3) ->
-                member.selectSingle(totem)
```
Comment (theQuazz) [inline wurst/systems/core/ClassSelector.wurst:143]: Why remove the selection of the totem? I thought that was kind of nice

Comment (totony-misc) [reply wurst/systems/core/ClassSelector.wurst:143]: good catch, forgot to add it back

Comment (totony-misc) [reply wurst/systems/core/ClassSelector.wurst:143]: idk why there's doAfter() here but it doesn't seem needed so I removed

File: wurst/objects/units/ShopObjects.wurst [hunk 3]
```diff
@@ -2,35 +2,64 @@ package ShopObjects
...
+@compiletime function createTotem()
+    let ABILITY_SELECT_UNITS_TOTEM = compiletime(UNIT_ID_GEN.next())
+
+    new AbilityDefinitionAlliedBuilding(ABILITY_SELECT_UNITS_TOTEM)
+        ..setActivationRadius(1, 1) // Small activation radius so it doesn't try selecting trolls
+        ..setShowSelectUnitButton(1, false)
+
+    new BuildingDefinition(UNIT_TROLL_TOTEM, UnitIds.cantina)
+        ..setName("Troll Totem")
+        ..setDescription("Troll picker")
```
Comment (jonathanmclaus) [inline wurst/objects/units/ShopObjects.wurst:53]: Does this have any functional purpose?

Comment (totony-misc) [reply wurst/objects/units/ShopObjects.wurst:53]: I actualyl dont know

Comment (jonathanmclaus) [reply wurst/objects/units/ShopObjects.wurst:53]: Didn't you add it? It wasn't in the raw definition.

Comment (totony-misc) [reply wurst/objects/units/ShopObjects.wurst:53]: I copied this from another BuildingDefinition iirc, I can remove if u want

Comment (Jaccouille) [reply wurst/objects/units/ShopObjects.wurst:53]: This field correspond to the text displayed when you hoover your icon over the building icon on your command button panel.

Comment (jonathanmclaus) [reply wurst/objects/units/ShopObjects.wurst:53]: I thought that required selecting multiple units though. You can't do that for the totem.

Comment (totony-misc) [reply wurst/objects/units/ShopObjects.wurst:53]: can't you just select the totem and your troll after you picked?

**Pull request merged**